5140 feilmeldinger i opprett sak

fredrik-oh opened the pull request:
https://jira.adeo.no/browse/MELOSYS-5140

Viser feilmeldinger fra backend i opprett ny sak.

Commit:
La til visning av feilmeldinger i opprettnysak.js

TODO: Konverter opprettnysak.js til tsx

Commit:
Refaktorering og typescripting av opprettnysak

Commit:
Merge branch 'master' into MELOSYS-5140_Feilmeldinger_i_opprett_sak

# Conflicts:
#	src/sider/opprettnysak/opprettnysak.less
#	src/sider/opprettnysak/opprettnysak.tsx

janitamarielle commented on src/ducks/lovvalgsperioder/types.test.js:
```diff
@@ -1,8 +1,8 @@
 
-describe("test lovvalgsperioder/types.js", () => {
-  describe("types.js", () => {
+describe("test lovvalgsperioder/types.ts", () => {
```
Hvorfor disse endringene? lovvalgsperioder/types.**ts** ser ut til å være types.**js**

fredrik-oh replied:
Merkelig, tror ikke jeg har endret dette 😮

janitamarielle replied:
Haha, kanskje IntelliJ har "hjulpet" deg når du renamet en annen types?

janitamarielle commented on src/ducks/vilkar/types.test.js:
```diff
@@ -1,8 +1,8 @@
 
-describe("test vilkar/types.js", () => {
-  describe("types.js", () => {
+describe("test vilkar/types.ts", () => {
```
Samme ☝️

janitamarielle commented on src/kodeverk/form.ts:
```diff
@@ -96,6 +96,18 @@ export const ARTIKKEL_16_1_VEDTAK = "artikkel_16_1_vedtak";
 export const ARTIKKEL_13_UTPEKLAND = "artikkel_13_utpekland";
 export const ARTIKKEL_13_X_VEDTAK = "artikkel_13_x_vedtak";
 export const OPPRETT_NY_SAK = "opprett_ny_sak";
+export interface OpprettNySakFormData {
```
Syns kanskje denne ble litt malplassert 🤔  Ser den kun blir brukt i opprettnysak.tsx. Kan den ligge der?

(Forsåvidt det samme med RegistreringPanelerFormData, som burde blitt flyttet til registrering)

janitamarielle commented on src/services/modules/oppgaver.js:
```diff
@@ -3,9 +3,9 @@ import { API_BASE_URL, OPPGAVER } from "../api-constants";
 export const oversikt = () => getAsJson(`${API_BASE_URL}${OPPGAVER}/oversikt`);
 
-export const sendPlukk = (oppgave) => postAsJson(`${API_BASE_URL}${OPPGAVER}/plukk`, oppgave);
+export const sendPlukk = (body) => postAsJson(`${API_BASE_URL}${OPPGAVER}/plukk`, body);
```
tror vi har brukt `data` mest i andre tilfeller, dersom du vil bytte fra oppgave til noe mer generelt.

janitamarielle commented on src/services/modules/types/oppgave.ts:
```diff
@@ -0,0 +1,11 @@
+import * as KV from "../../../kodeverk";
+
+export interface Oppgave {
```
Er dette typen til oppgave/body/data som sendes i requestene? => Gjøre modules/oppgaver.js om til modules/oppgaver.ts?

janitamarielle commented on src/sider/opprettnysak/opprettnysak.tsx:
```diff
@@ -22,38 +26,62 @@ import { erFeatureToggleEnabled } from "../../featuretoggle";
+import { feiletResponsSelectors } from "../../ducks/feiletRespons";
+import { Feilmeldinger } from "../../felleskomponenter/feilmeldinger";
+import { formOperations } from "../../ducks/form";
+import { OpprettReqDto } from "../../services/modules/fagsaker/fagsak";
```
Disse to kan blitt hentet ut fra Api.Oppgave....

janitamarielle commented on src/sider/opprettnysak/opprettnysak.tsx:
```diff
@@ -358,78 +442,30 @@ const OpprettNySak = ({ formValues, tilForsiden, handleSubmit, change, error, so
-    skalTilordnes: false,
-    behandlingstema: undefined,
-    soknadsinfo: { landkoder: [], erUkjenteEllerAlleEosLand: false },
-    hovedpart: BRUKER,
```
Denne må inn i initialValues

fredrik-oh replied:
Håndteres i destructuring nå L74-76
````   
const { behandlingstema, soknadsinfo, sakstype, hovedpart, brukerID, brukerNavn, virksomhetOrgnr, virksomhetNavn } =
    formValues || {};
const { landkoder, erUkjenteEllerAlleEosLand } = soknadsinfo || {};
````

janitamarielle replied:
Ser ikke hovedpart blir satt til BRUKER?

fredrik-oh replied:
Det har du rett i :) Fikser

janitamarielle commented on src/sider/opprettnysak/opprettnysak.tsx:
```diff
@@ -22,38 +26,62 @@ import { erFeatureToggleEnabled } from "../../featuretoggle";
+  const [oppgaver, setOppgaver] = useState<Oppgave[]>([]);
+  const [bekreftPending, setBekreftPending] = useState(false);
   const [oppgaverForsoktHentet, setOppgaverForsoktHentet] = useState(false);
+  const [folketrygdenToggleEnabled, setFolketrygdenToggleEnabled] = useState(false);
```
Er det en grunn til at du ikke setter denne direkte? Evt bruker `useFeatureToggle`

fredrik-oh replied:
Tar i bruk useFeatureToggle 👍

janitamarielle commented on src/sider/opprettnysak/opprettnysak.tsx:
```diff
@@ -22,38 +26,62 @@ import { erFeatureToggleEnabled } from "../../featuretoggle";
 
-  const folketrygdenToggleEnabled = erFeatureToggleEnabled("melosys.folketrygden.mvp");
-  const behandleAlleSakerToggleEnabled = erFeatureToggleEnabled("melosys.behandle_alle_saker");
+  const validerForm = () => {
```
Hvorfor trengs denne?

fredrik-oh replied:
Trigger visning av feilmeldinger i formen når saksbehandler trykker på opprett sak med ugyldig form state

janitamarielle replied:
Så visning av feil skal ikke skje før trykk?

fredrik-oh replied:
Nei, da ville vi vist feil på "alle" feltene før utfylling

janitamarielle commented on src/sider/opprettnysak/opprettnysak.tsx:
```diff
@@ -117,6 +154,40 @@ const OpprettNySak = ({ formValues, tilForsiden, handleSubmit, change, error, so
+    setBekreftPending(true);
+    const fom = soknadErValgt ? Utils.dato.formatterDatoTilISO(soknadsinfo.fom) : null;
+    const tomErUtfylt = soknadsinfo && soknadsinfo.tom;
+    const tom = tomErUtfylt && soknadErValgt ? Utils.dato.formatterDatoTilISO(soknadsinfo.tom) : null;
```
Dette er bare småtteri, men kan ikke 
```
const tomErUtfylt = soknadsinfo && soknadsinfo.tom;
const tom = tomErUtfylt && soknadErValgt ? Utils.dato.formatterDatoTilISO(soknadsinfo.tom) : null;
```

bli skrevet om til 

`const tom = soknadsinfo?.tom && soknadErValgt ? Utils.dato.formatterDatoTilISO(soknadsinfo.tom) : null;`

Commit:
Fikset en feilende test i personinfo.test.tsx

janitamarielle commented on src/sider/opprettnysak/opprettnysak.tsx:
```diff
@@ -333,18 +416,19 @@ const OpprettNySak = ({ formValues, tilForsiden, handleSubmit, change, error, so
                       avbrytTekst="Avbryt"
                       redigerbart
-                      bekreftRedigerbart={oppgaverFinnes}
+                      bekreftRedigerbart={!bekreftPending && oppgaverFinnes}
```
Bruke denne til å sette spinner og autoDisableWhenSpinner(eller noe)? Snakket med Cathrin og det var litt det som var best for UX.

janitamarielle commented on src/sider/opprettnysak/opprettnysak.tsx:
```diff
@@ -358,78 +442,30 @@ const OpprettNySak = ({ formValues, tilForsiden, handleSubmit, change, error, so
-    }
-  }
-};
+const { BRUKER, VIRKSOMHET } = MKV.Koder.aktoersroller;
```
Denne her nede? 🙃 Liker denne best øverst.. 🤔

fredrik-oh replied:
Foretrekker å samle const på samme sted. Og helst separere det fra redux. Kan stokke det om?

janitamarielle replied:
Kan være enig i å samle const. Personlig liker jeg at const/funksjoner jeg bruker i koden ligger lengre opp i filen enn der jeg bruker den. Dersom det er en spesifikk norm her så kan vi også følge den.

Commit:
QA feedback endringer

janitamarielle commented on src/ducks/lovvalgsperioder/types.test.js:
```diff
@@ -2,7 +2,7 @@
 
 describe("test lovvalgsperioder/types.js", () => {
-  describe("types.js", () => {
+  describe("types.ts", () => {
```
Ser du satte kommentaren til resolved, men denne er fortsatt feil i min nettleser?

fredrik-oh replied:
Jeg endret den til js, wtf 😕

janitamarielle committed:
QA som ble til litt mer:
erFeatureToggleEnabled er ett promise og man burde heller bruke useFeatureToggle i funksjonelle components der man ikke kan bruke await. Endrer derfor noen tilfeller nå.
Trekker IdentOgNavn ut av selve opprettnysak.tsx siden den blir buggy når den rerendres.

janitamarielle committed:
Småtteri i imports.
